rflx/expression.py

```diff
@@ -1265,15 +1314,43 @@ def __init__(
         negative: bool = False,
         immutable: bool = False,
         type_: rty.Type = rty.Undefined(),
+        argument_types: Sequence[rty.Type] = None,
         location: Location = None,
     ) -> None:
         self.name = ID(name)
         self.args = args or []
+        self.argument_types = argument_types or []
         super().__init__(negative, immutable, type_, location)
 
     def __neg__(self) -> "Call":
         return self.__class__(self.name, self.args, not self.negative)
 
+    def _check_type_subexpr(self) -> RecordFluxError:
+        error = RecordFluxError()
+
+        for a, t in itertools.zip_longest(self.args, self.argument_types):
+            if a:
+                error += a.check_type(t if t else rty.Any())
```

Comment:
That doesn't work. If `self.args` is shorter than `self.argument_types`, `a` would be a type (`rty.Any()`) while it must be an expression.

Reply:
That's why I used `self.argument_types[:len(self.args)]` which truncates `self.argument_types` to the length of `self.args`.

Reply:
I missed that one. Then it's a valid alternative. Maybe it's a bit faster because of fewer conditions. Changed in 2317ee40cdae0e927082bf4b98ad1cb2c6cc922c.